types/block.go

```diff
@@ -804,21 +815,29 @@ func (data *Data) StringIndented(indent string) string {
 		indent, data.hash)
 }
 
+// ToProto converts Data to protobuf
 func (data *Data) ToProto() *tmproto.Data {
+	if data == nil {
```

Comment:
@melekes @tess do these safety checks make sense to you? otherwise, the functions may panic

Reply:
Fine by me if data indeed can be nil at any point during the lifetime.

I'd prefer for us to use https://en.wikipedia.org/wiki/Null_Object_pattern, but it may not be possible here.